### PR TITLE
refactor: merge grpc_server_example

### PR DESCRIPTION
**Pull request by spacewander**
Like grpc-proxy & amesh, now we keep grpc_server_example to avoid
submitting multiple PRs.
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/grpc_server_example/main.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+		if err := stream.Send(&pb.HelloReply{Message: fmt.Sprintf("Hello %s", req.Name)}); err != nil {
+			return status.Errorf(codes.Unknown, "Failed to stream response back to client: %v", err)
+		}
+	}
+}
+
+func (s *server) Run(ctx context.Context, in *pb.Request) (*pb.Response, error) {
+	return &pb.Response{Body: in.User.Name + " " + in.Body}, nil
+}
```
Comment (bisakhmondal) [inline t/grpc_server_example/main.go:177]: Is there any utility of this RPC?

**Review by bisakhmondal — COMMENTED** [review]
In general, LGTM. Just curious, is there any specific use cases that were not being solved by the submodules? It's fine either way : )

**Comment by spacewander** [comment]
> In general, LGTM. Just curious, is there any specific use cases that were not being solved by the submodules? It's fine either way : )

Just tried to submit two PRs for one feature :)

File: t/grpc_server_example/main.go (hunk 1 above)
Comment (spacewander) [reply t/grpc_server_example/main.go:177]: It is used in https://github.com/apache/apisix/blob/6bb9feac824f0640e0e4ab799e37bbbf3adcdd96/t/plugin/grpc-transcode2.t#L264

Comment (bisakhmondal) [reply t/grpc_server_example/main.go:177]: Oh, I see. Thanks.

**Review by bisakhmondal — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**